app/src/main/java/com/brandon/forzenbook/viewmodels/ForzenTopLevelViewModel.kt

```diff
@@ -19,75 +20,81 @@ import javax.inject.Inject
         ) : LoginUiState
 
     }
 
-    fun dismissNotification() {
-        state.value = LoginUiState.PreCode
+    fun getCode(email: String) {
+        viewModelScope.launch {
+            state.value = LoginUiState.Loading
+            delay(5000)
+            if (email.isNotEmpty()) {
+                state.value = LoginUiState.PreCode(isEmailError = true)
+                cancel()
+            }
+            if (getLoginCodeUseCase(email)) {
+                state.value = LoginUiState.CodeSent
+            } else {
+                state.value = LoginUiState.PreCode(isGenericError = true)
+            }
+            Log.e("Brandon_Test_ViewModel","${state.value}")
+        }
     }
 
-    fun loginClicked(userName: String, code: String) {
+    fun loginClicked(email: String, code: String) {
         viewModelScope.launch {
             state.value = LoginUiState.Loading
-            loginUserName = userName
-            logincode = code
             delay(5000)
-            if (userName.isEmpty() || code.isEmpty()) {
-                state.value = LoginUiState.Error(
-                    isInvalidCredentialsError = true,
-                )
+            if (email.isEmpty() || code.isEmpty()) {
+                state.value = LoginUiState.PreCode(isEmailError = true)
                 cancel()
             }
-            val token = loginUseCase(userName, code)
+            val token = loginUseCase(email, code)
             if (token != null) {
-                if (token.token != null) {
-                    state.value = LoginUiState.PostCodeSent(loginToken = token.token)
-                } else {
-                    state.value = LoginUiState.Error(isGenericError = true)
-                }
+                    if (token.token != null) {
+                        state.value = LoginUiState.LoginWithCode(loginToken = token.token)
+                    }
+                    else { state.value = LoginUiState.Error(isGenericError = true) }
             }
-            Log.e(VIEWMODEL_ERROR_TAG, "${state.value}")
+            Log.e("Brandon_Test_ViewModel","${state.value}")
         }
     }
 
     fun isValidCode(code: String): Boolean {
         return code.isDigitsOnly()
     }
 
-    suspend fun isNetworkAvailable(context: Context?): Boolean {
+    fun isNetworkAvailable(context: Context?): Boolean {
         if (context == null) return false
-        val connectivityManager =
-            context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
+        val connectivityManager = context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
         if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) {
-            val capabilities =
-                connectivityManager.getNetworkCapabilities(connectivityManager.activeNetwork)
+            val capabilities = connectivityManager.getNetworkCapabilities(connectivityManager.activeNetwork)
             if (capabilities != null) {
```

Comment:
I don't believe you have to check for each of these 3 specifically. https://stackoverflow.com/questions/4238921/detect-whether-there-is-an-internet-connection-available-on-android

Reply:
after reading through that, you are 100% correct. I have removed the when block on the checks and am now simply checking to see if the device is connected to anything to see if they have internet connection.